<launch>
  <arg name="odom_topic" />

  <!-- Convert odometry to Pose2D -->
  <node pkg="kobuki_node" type="getOdom2D.py" name="getOdom2D">
    <remap from="odom" to="$(arg odom_topic)" />
    <remap from="pose2d" to="/pose2d" />
  </node>

  <!-- Control node -->
  <node pkg="base_navigation" type="control.py" name="control" output="screen">
    <param name="turn_tolerance" value="0.01" />
    <param name="turn_speed" value="1" />
    <param name="move_speed" value="0.7" />
    <remap from="pose2d" to="/pose2d" />
    <remap from="odom" to="$(arg odom_topic)" />
    <remap from="cmd_vel" to="/control/cmd_vel_raw" />
  </node>

  <!-- Velocity smoother -->
  <node pkg="nodelet" type="nodelet" name="velocity_smoother_nodelet_manager" args="manager" />
  <node pkg="nodelet" type="nodelet" name="velocity_smoother"
        args="load yocs_velocity_smoother/VelocitySmootherNodelet velocity_smoother_nodelet_manager">
    <param name="accel_lim_v" value=".5" />
    <param name="accel_lim_w" value="1" />
    <param name="speed_lim_v" value=".7" />
    <param name="speed_lim_w" value="1.9" />
    <param name="robot_feedback" value="1" />
    <remap from="velocity_smoother/raw_cmd_vel" to="/control/cmd_vel_raw" />
    <remap from="velocity_smoother/odometry" to="/$(arg odom_topic)" />
    <remap from="velocity_smoother/smooth_cmd_vel" to="/cmd_vel_mux/input/teleop" />
  </node>
</launch>
